### PR TITLE
Implement context menu events for AVM2

### PR DESCRIPTION
**Pull request by Lord-McSweeney**
This allows you to complete Stage 7 in This Is The Only Level 3.

Unfortunately doesn't help with #10493.

### CODE REVIEW

File: core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as [hunk 1]
```diff
@@ -1,13 +1,97 @@
...
+        private var _save:Boolean = true;
+        private var _zoom:Boolean = true;
+        
+        public function get forwardAndBack():Boolean {
+            return this._forwardAndBack;
+        }
+        
+        public function set forwardAndBack(value:Boolean):void {
+            this._forwardAndBack = value;
+        }
```
Comment (Dinnerbone) [inline core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as:19]: Why did you convert from public fields to trivial getter/setters? Here and elsewhere

Comment (Lord-McSweeney) [reply core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as:19]: This matches FP. I know it's kinda weird, but there's no harm in doing it, and this way, `describeType` output will match as well.

**Review by Dinnerbone — APPROVED** [review]

File: core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as [hunk 2]
```diff
@@ -1,13 +1,97 @@
...
-		public var loop:Boolean = true;
-		public var play:Boolean = true;
-		public var print:Boolean = true;
-		public var quality:Boolean = true;
-		public var rewind:Boolean = true;
-		public var save:Boolean = true;
-		public var zoom:Boolean = true;
-	}
+    public final class ContextMenuBuiltInItems {
+        private var _forwardAndBack:Boolean = true;
```
Comment (adrian17) [inline core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as:4]: a bit late but: was there a reason to convert these?
I'm tempted to convert back ;D

File: core/src/player.rs [hunk 3]
```diff
@@ -679,8 +698,40 @@ impl Player {
...
+                            let mut activation = Avm2Activation::from_nothing(context.reborrow());
+
+                            let menu_obj = root_dobj
+                                .as_interactive()
+                                .map(|iobj| iobj.context_menu())
+                                .and_then(|v| v.as_object());
+
+                            if menu_obj.is_some() {
+                                // TODO: contextMenuOwner and mouseTarget might not be the same (see above comment)
+                                let menu_evt = activation
```
Comment (adrian17) [inline core/src/player.rs:716]: all of this should have been extracted to its own method, just like `run_context_menu_custom_callback` for AVM1

File: desktop/src/gui.rs [hunk 4]
```diff
@@ -410,7 +410,9 @@ impl RuffleGui {
                     let clicked = if item.checked {
                         Checkbox::new(&mut true, &item.caption).ui(ui).clicked()
```
Comment (adrian17) [inline desktop/src/gui.rs:411]: this doesn't appear to support checked items?

File: core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as (hunk 2 above)
Comment (Lord-McSweeney) [reply core/src/avm2/globals/flash/ui/ContextMenuBuiltInItems.as:4]: These are technically correct to flash (see the above review by Dinnerbone). I don't know why Flash does it this way.

**Pull request merged**